sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/main/java/com/azure/messaging/eventhubs/checkpointstore/blob/BlobCheckpointStore.java

```diff
@@ -293,60 +414,84 @@ private Mono<Void> reportMetrics(Mono<Void> checkpointMono, Checkpoint checkpoin
             });
     }
 
-    private String getBlobPrefix(String fullyQualifiedNamespace, String eventHubName, String consumerGroupName,
-        String typeSuffix) {
-        return fullyQualifiedNamespace + BLOB_PATH_SEPARATOR + eventHubName + BLOB_PATH_SEPARATOR + consumerGroupName
-            + typeSuffix;
+    private static boolean isBlobNotFoundException(Throwable error, String partitionId) {
+        if (!(error instanceof BlobStorageException)) {
+            LOGGER.atVerbose()
+                .addKeyValue(PARTITION_ID_LOG_KEY, partitionId)
+                .log(Messages.CLAIM_ERROR + ". Not BlobStorageException.", error);
+            return false;
+        } else if (((BlobStorageException) error).getErrorCode() != BlobErrorCode.BLOB_NOT_FOUND) {
+            LOGGER.atVerbose()
+                .addKeyValue(PARTITION_ID_LOG_KEY, partitionId)
+                .log(Messages.CLAIM_ERROR + ". Not BlobErrorCode.BLOB_NOT_FOUND.", error);
+            return false;
+        } else {
+            return true;
+        }
     }
 
-    private String getBlobName(String fullyQualifiedNamespace, String eventHubName, String consumerGroupName,
-        String partitionId, String typeSuffix) {
-        return fullyQualifiedNamespace + BLOB_PATH_SEPARATOR + eventHubName + BLOB_PATH_SEPARATOR + consumerGroupName
-            + typeSuffix + partitionId;
+    private static String getBlobPrefix(String fullyQualifiedNamespace, String eventHubName, String consumerGroupName,
+        String typeSuffix, boolean isLowercase) {
+
+        final String prefix = fullyQualifiedNamespace + BLOB_PATH_SEPARATOR + eventHubName + BLOB_PATH_SEPARATOR
+            + consumerGroupName + typeSuffix;
```

Comment:
Are all these strings static for the lifetime of this object? If so, could we just create the cased and lowercased versions of these strings at construction time.